### PR TITLE
Additional unicode fixes

### PR DESCRIPTION
**Pull request by roee88**
Fixes for custom thumbnail,icon,background paths with unicode characters.

### CODE REVIEW

File: resources/lib/gui.py [hunk 1]
```diff
@@ -473,6 +473,12 @@ def _save_shortcuts( self, weEnabledSystemDebug = False, weEnabledScriptDebug =
                 # Inform user menu couldn't be saved
                 xbmcgui.Dialog().ok( __addon__.getAddonInfo( "name" ), __language__( 32097 ), __language__( 32094 ) )                 
 
+    def _try_decode(self, text, encoding="utf-8"):
+        try:
+            return text.decode(encoding)
+        except:
+            return text
```
Comment (phil65) [inline resources/lib/gui.py]: it would probably make more sense to put this into a separate module instead of adding it to gui class?

**Comment by ghost** [comment]
Not sure I agree with adding a module for a single function (but it's not rare for phil65 and I to disagree on coding methodologies ;)), otherwise the changes look good. @phil65, if you're happy with the new module then I think this can be merged.

And thanks for the fixes.

File: resources/lib/datafunctions.py [hunk 2]
```diff
@@ -1107,10 +1102,8 @@ def upgrade_xmlfile( self, path, mixedVersion = False, saveFile = True ):
             actionTree = xmltree.SubElement( root, "shortcut" )
             
             # Action
-            try:
-                action = urllib.unquote( shortcut[4] ).decode( "utf-8" )
-            except:
-                action = urllib.unquote( shortcut[4] )
+            action = try_decode( urllib.unquote( shortcut[4] ) )
```
Comment (ghost) [inline resources/lib/datafunctions.py:1105]: Just a note, the upgrade_\* functions will be dropped when #30 is merged :)

**Comment by phil65** [comment]
I doesnt have to be a new module, but since that method does not really do anything concerning the state of the class (=it doesnt need self) it should either be static or get moved out to some module with helper functions.

**Comment by ghost** [comment]
Now there we agree ;) My instinct was it would fit nicely in datafunctions.py (which actually isn't static/a static class/a module because this script is really badly coded on so many levels, but it at least does have a lot of helper functions), which every other class/module/whatever has an instance of, and is used for bringing most data into the script and checking it - possibly somewhere around the local() code, which is very broadly similar.

It's also good to see how moving it out of gui.py meant it could be used in so many more situations :)

Edit: Actually this might be a good opportunity to lay the groundwork as it were for moving functions out of classes and into modules, if instead this function was in a more generically named module. There's a lot of code, particularly in datafunctions.py, which could very easily be - and in the long-run SHOULD be - a function in a module, not a class.

**Comment by marcelveldt** [comment]
@Ignoble61 will you do the merge if everything is fine here ?

**Comment by ghost** [comment]
Fundamentals here are great, and the try_decode() can be moved somewhere more suitable (or unicodeutils can be be made more useful) in the future, so let's get this merged. Thanks again for the fixes, @roee88

**Pull request merged**